src/tunnel/server.rs

```diff
@@ -58,28 +58,32 @@ async fn run_tunnel(
 ) -> anyhow::Result<(RemoteAddr, Pin<Box<dyn AsyncRead + Send>>, Pin<Box<dyn AsyncWrite + Send>>)> {
     match remote.protocol {
         LocalProtocol::Udp { timeout, .. } => {
-            let (rx, tx) = UdpTunnelConnector::new(
+            let connector = UdpTunnelConnector::new(
                 &remote.host,
                 remote.port,
                 server_config.socket_so_mark,
                 timeout.unwrap_or(Duration::from_secs(10)),
                 &server_config.dns_resolver,
-            )
-            .connect(&None)
-            .await?;
+            );
+            let (rx, tx) = match &server_config.http_proxy {
+                None => connector.connect(&None).await?,
+                Some(proxy_url) => connector.connect_with_http_proxy(proxy_url, &None).await?,
```

Comment:
You cant forward udp traffic with an HTTP proxy. It is going to work only for TCP based protocol.